modules/c/pages/working-with-vector-search.adoc

```diff
@@ -0,0 +1,231 @@
+
+|Expression
+|Required
+|No default
+|A SQL++ expression indicating where to get the vectors. 
+A document property for embedded vectors or
+`prediction()` to call a registered Predictive model.
+|Expression Language
+|Required
+|No default
+|Specifies the query language used. This can be N1Q1 or JSON query.
+For JSON query syntax, see link:https://github.com/couchbase/couchbase-lite-core/wiki/JSON-Query-Schema[Couchbase JSON query wiki].
+|Number of Dimensions
+|Required
+|No default
+|2-2048
+|Number of Centroids
+|Required
+|No default
+|1-64000. The general guideline is an approximate square root of the number of documents
+|Distance Metric
+|Optional
+|Euclidean Distance
+|You can also configure Cosine Distance as your Distance Metric
+|Encoding
+|Optional
+| Scalar Quantizer(SQ) or SQ-8 bits
+a|There are three possible configurations:
+
+* None
+No compression, No data loss
+* Scalar Quantizer (SQ) or SQ-8 bits (Default)
+Reduces the number of bits per dimension 
+* Product Quantizer (PQ)
+Reduces the number of dimensions and bits per dimension
+
+|Training Size
+|Optional
+a|There are defaults for both the minimum and maximum training size for training the vectors
+
+* The minimum training size is set to 25x the number of Centroids
+
+* The maximum training size is set to 256x the number of Centroids
+|
+
+|===
+
+CAUTION: Altering the default training sizes could be detrimental to the accuracy of returned results produced by the model and total computation time.
+
+== Generating Vectors
+
+You can use two methods to generate vectors in Couchbase Lite:
+
+. You can call a Machine Learning(ML) model, and embed the generated vectors inside the documents.
+
+. You can use the `prediction()` function to generate vectors to be indexed for each document at the indexing time.
+
+Below are example configurations of the previously mentioned methods.
+
+=== Create a Vector Index with Embeddings
+
+This method shows you how to create a Vector Index with embeddings.
+
+[source, c]
+----
+
+include::c:example$code_snippets/VectorSearch.h[tags=vs-create-index]
+
+----
+
+. First, create the standard configuration, setting up an expressions, number of dimensions and number of centroids for the vector embedding.
+
+. Next, get the colors collection from the database.
+
+. Finally, create a vector index, colors_index, on the collection with the configuration.
+
+=== Create Vector Index Embeddings from a Predictive Model
+
+This method generates vectors to be indexed for each document at the index time by using the `prediction()` function.
+
+[source, c]
+----
+
+include::c:example$code_snippets/VectorSearch.h[tags=vs-create-predictive-index]
+
+----
+
+NOTE: You can use less storage by using the `prediction()` function as the encoded vectors will only be stored in the index. 
+However, the index time will be longer as vector embedding generation is occurring at run time.
+
+== Vector SQL++ Functions
```

Comment:
Check this renders properly, if not you may need to use `{sqlpp}`

Reply:
Yeah it's all fine